packages/generate/src/edgeql-js/generateCastMaps.ts

```diff
@@ -62,10 +62,25 @@ export const generateCastMaps = (params: GeneratorParams) => {
 
   const staticMap = new CodeBuffer();
   staticMap.writeln([dts`declare `, t`type getSharedParentScalar<A, B> =`]);
+  staticMap.writeln([t`  IsEqual<A, B> extends true ? A :`]);
+
   const runtimeMap = new CodeBuffer();
 
   const returnTypes = new Set<string>();
 
+  const uncastables = new Set(
```

Comment:
Like uncrustables but saltier.
Open to different names.

Reply:
Nope, this is the perfect name 👨‍🍳